src/components/instant.rs

```diff
@@ -278,35 +270,240 @@ impl Instant {
 /// Utility for determining if the nanos are within a valid range.
 #[inline]
 #[must_use]
-pub(crate) fn is_valid_epoch_nanos(nanos: &BigInt) -> bool {
-    nanos <= &BigInt::from(crate::NS_MAX_INSTANT) && nanos >= &BigInt::from(crate::NS_MIN_INSTANT)
+pub(crate) fn is_valid_epoch_nanos(nanos: &i128) -> bool {
+    (crate::NS_MIN_INSTANT..=crate::NS_MAX_INSTANT).contains(nanos)
+}
+
+impl FromStr for Instant {
+    type Err = TemporalError;
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let ixdtf_record = parse_instant(s)?;
+
+        // NOTE: parse_instant rejects on missing required fields. So the below
+        // `temporal_unwraps` MUST be present.
+
+        // Find the IsoDate
+        let date = ixdtf_record.date.temporal_unwrap()?;
```

Comment:
Hmm, is this asserting an invariant? Because `temporal_unwrap` should only be used to ensure invariants are preserved. If this is used to unwrap error caused by the user, then I think it should return an error instead, which is a bit more explanatory.

Reply:
It should be an invariant at this point. `parse_instant` requires a string with a `DateRecord`, `TimeRecord`, and `OffsetRecord`, and throws an error if any of them are `None`.

We can return an error in these places again, but they would be redundant errors that realistically would never throw.

Reply:
Oh, now I see why. On `parse_instant`, we validate that the fields are `Some` and throw otherwise. Can we move that invariant to the typesystem? Instead of returning an `IxdtfRecord`, we return `IxdtfDurationRecord` with the validated fields being not `Option`.